agr_literature_service/api/crud/topic_entity_tag_crud.py

```diff
@@ -0,0 +1,168 @@
+        details = "One of the XXXX_entity's MUST be set"
+        okay = False
+    elif count > 1:
+        details = "ONLY one of the XXXX_entity's MUST be set"
+        okay = False
+    return (okay, details)
+
+
+def create(db: Session, topic_entity_tag: TopicEntityTagSchemaCreate) -> int:
+    """
+    Create a new topic_entity_tag
+    :param db:
+    :param topic_entity_tag:
+    :return:
+    """
+
+    topic_entity_tag_data = jsonable_encoder(topic_entity_tag)
+    add_default_create_keys(db, topic_entity_tag_data)
+
+    reference_curie = topic_entity_tag_data["reference_curie"]
+    del topic_entity_tag_data["reference_curie"]
+    reference = db.query(ReferenceModel).filter(ReferenceModel.curie == reference_curie).first()
+    if not reference:
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                            detail=f"Reference with curie {reference_curie} does not exist")
+    topic_entity_tag_data["reference_id"] = reference.reference_id
+
+    (okay, details) = extra_checks(topic_entity_tag_data)
+    if not okay:
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                            detail=details)
+
+    props = []
+    if "props" in topic_entity_tag_data and topic_entity_tag_data["props"]:
+        props = topic_entity_tag_data["props"]
+        del topic_entity_tag_data["props"]
+    elif "props" in topic_entity_tag_data:
+        del topic_entity_tag_data["props"]
+
+    db_obj = TopicEntityTagModel(**topic_entity_tag_data)
+    db.add(db_obj)
+    db.commit()
+    for prop in props:
+        xml = {"topic_entity_tag_id": db_obj.topic_entity_tag_id,
+               "qualifier": prop['qualifier'],
+               "created_by": topic_entity_tag_data["created_by"]}
+        add_default_create_keys(db, xml)
+        prop_obj = TopicEntityTagPropModel(**xml)
+        db.add(prop_obj)
+    db.commit()
+    return db_obj.topic_entity_tag_id
+
+
+def show(db: Session, topic_entity_tag_id: int):
+    """
+
+    :param db:
+    :param topic_entity_tag_id:
+    :return:
+    """
+
+    topic_entity_tag = db.query(TopicEntityTagModel).filter(TopicEntityTagModel.topic_entity_tag_id == topic_entity_tag_id).first()
+    if not topic_entity_tag:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                            detail=f"topic_entityTag with the topic_entity_tag_id {topic_entity_tag_id} is not available")
+
+    topic_entity_tag_data = jsonable_encoder(topic_entity_tag)
+
+    if topic_entity_tag_data["reference_id"]:
+        topic_entity_tag_data["reference_curie"] = db.query(ReferenceModel).filter(ReferenceModel.reference_id == topic_entity_tag_data["reference_id"]).first().curie
+        del topic_entity_tag_data["reference_id"]
+
+    props = db.query(TopicEntityTagPropModel).filter(TopicEntityTagPropModel.topic_entity_tag_id == topic_entity_tag_id).all()
+    topic_entity_tag_data["props"] = []
+    prop: TopicEntityTagPropModel
```

Comment:
What is this line for ?  I'm not familiar with the syntax, and in the next line prop is each value of props

Reply:
mypy. It makes mypy happy and editor knows what it is so can check methods etc.